synapse_data_warehouse/synapse/dynamic_tables/V2.29.4__userprofile_latest.sql

```diff
@@ -0,0 +1,49 @@
+-- Introduce the dynamic table
+USE SCHEMA {{database_name}}.synapse; --noqa: JJ01,PRS,TMP
+CREATE OR REPLACE DYNAMIC TABLE USERPROFILE_LATEST
+(
+	CHANGE_TYPE VARCHAR(16777216) COMMENT 'The type of change that occurred to the user profile, e.g., CREATE, UPDATE (Snapshotting does not capture DELETE change).',
+	CHANGE_TIMESTAMP TIMESTAMP_NTZ(9) COMMENT 'The time when any change to the user profile was made (e.g. create or update).',
+	CHANGE_USER_ID NUMBER(38,0) COMMENT 'The unique identifier of the user who made the change to the user profile.',
+	SNAPSHOT_TIMESTAMP TIMESTAMP_NTZ(9) COMMENT 'The time when the snapshot was taken (It is usually after the change happened).',
+	ID NUMBER(38,0) COMMENT 'The unique identifier of the user.',
+	USER_NAME VARCHAR(16777216) COMMENT 'The Synapse username.',
+	FIRST_NAME VARCHAR(16777216) COMMENT 'The first name of the user.',
+	LAST_NAME VARCHAR(16777216) COMMENT 'The last name of the user.',
+	EMAIL VARCHAR(16777216) COMMENT 'The primary email of the user.',
+	SNAPSHOT_DATE DATE COMMENT 'The data is partitioned for fast and cost effective queries. The snapshot_timestamp field is converted into a date and stored in the snapshot_date field for partitioning. The date should be used as a condition (WHERE CLAUSE) in the queries.',
+	CREATED_ON TIMESTAMP_NTZ(9) COMMENT 'The creation time of the user profile.',
+	IS_TWO_FACTOR_AUTH_ENABLED BOOLEAN COMMENT 'Indicates if the user had two factor authentication enabled when the snapshot was captured.',
+    TOS_AGREEMENTS VARIANT COMMENT 'Contains the list of all the term of service that the user agreed to, with their agreed on date and version.',
+    LOCATION VARCHAR(16777216) COMMENT 'The location of the user.',
+	COMPANY VARCHAR(16777216) COMMENT 'The company where the user works.',
+	POSITION VARCHAR(16777216) COMMENT 'The position of the user in the company.',
+	INDUSTRY VARCHAR(16777216) COMMENT 'The industry/discipline that this person is associated with.'
```

Comment:
I wonder if explicitly listing out the columns like this would ever clash with the `SELECT *` from the main query, if say `userprofilesnapshot` gets a column added to it later down the line? So either we keep the `SELECT *` and implicitly call the columns in this script + introduce the column comments in a separate script, or the columns selected are explicitly written out in the main query of this script

Reply:
If you're not modifying any columns from the upstream table I recommend using the more succinct style which you can see in Jenny's PR which I linked to below.

Reply:
Good point. I will add column comments as a separate step as like what Jenny said, new columns might be added to the source table.

Reply:
Just a heads up: I noticed that dynamic table comments don't show up in their own little Description box the way regular table comments do. With the dynamic table, you have to dive into the **Table Definition** itself to see that the dynamic table comment was added. It would be nice if we can also have the dynamic table comments front and center the way it is with the normal ones, but for now this is what it is. Dynamic tables were just added last year, so I'm sure this is a QOL improvement they'll add in the future.

Reply:
Yes, the only ways to check the Table comments are either hover over the table name and the comment section which is pretty tiny or refer to the Table Definition. Hope the Description box gets added soon.